src/hooks/ens.ts

```diff
@@ -31,15 +31,15 @@ export interface EnsResolver {
 export const useEnsResolver: () => EnsResolver = () => {
   const { safe, sdk } = useSafeAppsSDK();
   const web3Provider = useMemo(() => new ethers.providers.Web3Provider(new SafeAppProvider(safe, sdk)), [sdk, safe]);
-
   const resolveCache = useMemo(() => new Map<string, string | null>(), []);
 
   const lookupCache = useMemo(() => new Map<string, string | null>(), []);
 
   const cachedResolveName = useCallback(
     async (ensName: string) => {
       const cachedAddress = resolveCache.get(ensName);
-      const resolvedAddress = cachedAddress ? cachedAddress : await web3Provider.resolveName(ensName);
+      const resolvedAddress =
+        typeof cachedAddress !== "undefined" ? cachedAddress : await web3Provider.resolveName(ensName);
```

Comment:
Looks like introducing the tests revealed a potential bug here... or was this just an adaption to accommodate the test itself?

Reply:
null was not cached. So names which can't be resolved where retried and not cached.